src/Cassandra/Requests/RequestExecution.cs

```diff
@@ -389,7 +389,7 @@ private void HandleRequestError(IRequestError error, Host host)
                     break;
             }
 
-            _requestObserver.OnRequestError(host, retryInformation.Reason, retryInformation.Decision.DecisionType);
+            _requestObserver.OnNodeRequestError(host, retryInformation.Reason, retryInformation.Decision.DecisionType);
```

Comment:
By the way, you can add the `OnNodeSuccess` call at the top of the `private void HandleRowSetResult(Response response)` method and with this we will have all the request tracker methods implemented in the core driver 👍

Reply:
@joao-r-reis , I implemented this as you suggested and I also included a method named `OnNodeStart()` in the `RequestExecution.Start()`. 

The idea is to have nested activities for node level requests (that will appear inside the session level requests) that will give the user information regarding the host and execution retries.
As we already had the session start, error, and finish implementations, I think it didn't make much sense to have only the error and finish implementations for node requests as, for activities, there was nothing being started. With that in mind, the activity names are now called `Session Request` and `Node Request` instead of just `Request` as you suggested here.
https://github.com/datastax/csharp-driver/pull/608#issuecomment-2122837103

What do you think about this?

Reply:
Makes perfect sense 👍 I like it.